pkg/apis/serving/k8s_validation.go

```diff
@@ -82,10 +82,18 @@ var (
 )
 
 // ValidateVolumes validates the Volumes of a PodSpec.
-func ValidateVolumes(vs []corev1.Volume, mountedVolumes sets.String) (sets.String, *apis.FieldError) {
+func ValidateVolumes(ctx context.Context, vs []corev1.Volume, mountedVolumes sets.String) (sets.String, *apis.FieldError) {
 	volumes := make(sets.String, len(vs))
 	var errs *apis.FieldError
+	features := config.FromContextOrDefaults(ctx).Features
 	for i, volume := range vs {
+		// skip validating empty dir volumes if the feature is not enabled
```

Comment:
Im not sure I understand this comment; we skip validating if the feature (that is, allowing this type of volume) is enabled. (If the feature is disabled then we DO validate that the volume is not there). Is the comment supposed to be further down maybe, above the validateEmptyDirFields line?

Reply:
Yeah this needs to be removed initially I had a different logic here.